servicex/transformer/servicex_adapter.py

```diff
@@ -27,17 +27,40 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 import datetime
 import requests
+import os
+from urllib3.util.retry import Retry
+from requests.adapters import HTTPAdapter
 
 
 class ServiceXAdapter:
     def __init__(self, servicex_endpoint):
         self.server_endpoint = servicex_endpoint
+        self.session = requests.session()
 
-    def post_status_update(self, status_msg):
-        requests.post(self.server_endpoint + "/status", data={
-            "timestamp": datetime.datetime.now().isoformat(),
-            "status": status_msg
-        })
+        retries = Retry(total=5,
+                        connect=3,
+                        backoff_factor=0.1)
+        self.session.mount('http://', HTTPAdapter(max_retries=retries))
+
+    def post_status_update(self, file_id, status_code, info):
+        success = False
+        attempts = 0
+        while not success and attempts < 3:
```

Comment:
Why isn't this also using the "retry" package?

Reply:
I'm thinking of the Retry package: https://pypi.org/project/retry/

Reply:
Thanks - I didn't know about that. Much better! The power of code reviews.